Pages/Incoming.cshtml.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Octokit;
+
+namespace DependencyFlow.Pages
+{
+    public class IncomingModel : PageModel
+    {
+        private static readonly Regex _repoParser = new Regex(@"https?://(www\.)?github.com/(?<owner>[A-Za-z0-9-_\.]+)/(?<repo>[A-Za-z0-9-_\.]+)");
+        private readonly swaggerClient _client;
+        private readonly GitHubClient _github;
+
+        public IncomingModel(swaggerClient client, GitHubClient github)
+        {
+            _client = client;
+            _github = github;
+        }
+
+        public IReadOnlyList<IncomingRepo> IncomingRepositories { get; private set; }
+        public RateLimit CurrentRateLimit { get; private set; }
+
+        public async Task OnGet(int channelId, string repo)
+        {
+            // The 'repo' is URL encoded (because it's a URL within a URL). Decode it before calling the API.
+            var decodedRepo = WebUtility.UrlDecode(repo);

```

Comment:
We're hoping to add an attribute that does this for you in 5.0 😆

Reply:
I'm 100% fine with doing this, but an attribute would be niceness

Reply:
Yeah, it's a case of old school vs new school.

The fact that we don't/can't know when to unescape slashes for you catches people out. If you have a feature for it, then it seems like we're setting users up for success.  It comes up A LOT.